File: CourseWork_Graphical_interface/CourseWork_Graphical_interface/Form1.cs
//Программа для вычисления площади фигуры, ограниченной тремя кривыми
//Программист Летов Н.Е.

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using CourseWorkDLL;

namespace CourseWork_Graphical_interface
{
    public partial class Form1 : Form
    {
        expressions box = new expressions();
        point[] Cross_Points_1_2, Cross_Points_2_3, Cross_Points_1_3;
        value_with_error square;

        public Form1()
        {
            InitializeComponent();
        }

        private void exit_button_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void next_button_1_Click(object sender, EventArgs e)
        {
            box.a = double.Parse(a_value_box.Text);
            box.b = double.Parse(b_value_box.Text);
            box.c = double.Parse(c_value_box.Text);

            back_button_1.Enabled = true;
            next_button_1.Visible = false;
            next_button_2.Visible = true;
            parametries_title_label.Visible = false;
            A_value_label.Visible = false;
            a_value_box.Visible = false;
            b_value_label.Visible = false;
            b_value_box.Visible = false;
            c_value_label.Visible = false;
            c_value_box.Visible = false;
            limit_title_label.Visible = true;
            left_limit_label.Visible = true;
            left_limit_box.Visible = true;
            right_limit_label.Visible = true;
            right_limit_box.Visible = true;
        }

        private void back_button_1_Click(object sender, EventArgs e)
        {
            error_label.Visible = false;
            back_button_1.Enabled = false;
            next_button_1.Visible = true;
            next_button_2.Visible = false;
            parametries_title_label.Visible = true;
            A_value_label.Visible = true;
            a_value_box.Visible = true;
            b_value_label.Visible = true;
            b_value_box.Visible = true;
            c_value_label.Visible = true;
            c_value_box.Visible = true;
            limit_title_label.Visible = false;
            left_limit_label.Visible = false;
            left_limit_box.Visible = false;
            right_limit_label.Visible = false;
            right_limit_box.Visible = false;
        }

        private void next_button_2_Click(object sender, EventArgs e)
        {
            if (!Cross_Points.Check_Limits(double.Parse(left_limit_box.Text),double.Parse(right_limit_box.Text)))
                error_label.Visible = true;
            else
            {
                error_label.Visible = false;
                limit_title_label.Visible = false;
                left_limit_label.Visible = false;
                left_limit_box.Visible = false;
                right_limit_label.Visible = false;
                right_limit_box.Visible = false;
                back_button_2.Visible = true;
                fault_label.Visible = true;
                eps_label.Visible = true;
                eps_box.Visible = true;
                partition_label.Visible = true;
                partition_box.Visible = true;
                next_button_2.Visible = false;
                calculate_button.Visible = true;
                back_button_1.Visible = false;
            }
        }

        private void back_button_2_Click(object sender, EventArgs e)
        {
            error_label.Visible = false;
            limit_title_label.Visible = true;
            left_limit_label.Visible = true;
            left_limit_box.Visible = true;
            right_limit_label.Visible = true;
            right_limit_box.Visible = true;
            back_button_2.Visible = false;
            parametries_title_label.Visible = false;
            eps_label.Visible = false;
            eps_box.Visible = false;
            partition_label.Visible = false;
            partition_box.Visible = false;
            next_button_2.Visible = true;
            calculate_button.Visible = false;
            back_button_1.Visible = true;
            fault_label.Visible = false;
        }

        private void calculate_button_Click(object sender, EventArgs e)
        {
            if (!(Square.Check_Partition(int.Parse(partition_box.Text)) && Cross_Points.Check_Epsilon(double.Parse(eps_box.Text)))) error_label.Visible = true;
            else
            {
                Cross_Points_1_2 = Cross_Points.Find_Cross_Points(double.Parse(left_limit_box.Text),
                                                                  double.Parse(right_limit_box.Text),
                                                                  double.Parse(eps_box.Text), box.F1, box.F2);

                Cross_Points_2_3 = Cross_Points.Find_Cross_Points(double.Parse(left_limit_box.Text),
                                                                  double.Parse(right_limit_box.Text),
                                                                  double.Parse(eps_box.Text), box.F2, box.F3);

                Cross_Points_1_3 = Cross_Points.Find_Cross_Points(double.Parse(left_limit_box.Text),
                                                                  double.Parse(right_limit_box.Text),
                                                                  double.Parse(eps_box.Text), box.F1, box.F3);
                error_label.Visible = false;
                back_button_2.Visible = false;
                calculate_button.Visible = false;
                print_points_button.Visible = true;
                eps_label.Visible = false;
                eps_box.Visible = false;
                partition_label.Visible = false;
                partition_box.Visible = false;
                fault_label.Visible = false;
            }
        }
        private void print_points_button_Click(object sender, EventArgs e)
        {
            cross_point_label.Visible = true;
            hint_1_2_label.Visible = true;
            hint_1_3_label.Visible = true;
            hint_2_3_label.Visible = true;
            print_1_2_label.Text = String.Format("({0:f5},{1:f5})",Cross_Points_1_2[0].x,Cross_Points_1_2[0].y);
            print_1_2_label.Visible = true;
            print_2_3_label.Text = String.Format("({0:f5},{1:f5})", Cross_Points_2_3[0].x, Cross_Points_2_3[0].y);
            print_2_3_label.Visible = true;
            print_1_3_label.Text = String.Format("({0:f5},{1:f5})", Cross_Points_1_3[0].x, Cross_Points_1_3[0].y);
            print_1_3_label.Visible = true;
            print_points_button.Visible = false;
            area_button.Visible = true;
            
        }

        private void area_button_Click(object sender, EventArgs e)
        {
            cross_point_label.Visible = false;
            hint_1_2_label.Visible = false;
            hint_1_3_label.Visible = false;
            hint_2_3_label.Visible = false;
            print_1_2_label.Visible = false;
            print_2_3_label.Visible = false;
            print_1_3_label.Visible = false;
            square = Square.Final_Square(Cross_Points_2_3[0].x, Cross_Points_1_2[0].x, Cross_Points_1_3[0].x,
                                                          box.F2, box.F1, box.F3, double.Parse(eps_box.Text) , int.Parse(partition_box.Text));
            
            area_button.Visible = false;
            print_area_button.Visible = true;
        }

        private void print_area_button_Click(object sender, EventArgs e)
        {
            area_label.Text = String.Format("Площадь фигуры равна {0:f8},\nпри погрешности {1:f8}", square.value, square.error);
            area_label.Visible = true;
            print_area_button.Visible = false;
        }
    }
}

File: CourseWork_Graphical_interface_VB/CourseWork_Graphical_interface_VB/Form1.vb
'Программа для вычисления площади фигуры, ограниченной тремя кривыми
'Программист Летов Н.Е.

Imports CourseWorkDLL

Public Class Form1

    Public box = New expressions()
    Public Cross_Points_1_2, Cross_Points_2_3, Cross_Points_1_3 As point()
    Public plaza = New value_with_error()


    Private Sub exit_button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles exit_button.Click
        Close()
    End Sub

    Private Sub next_button_1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles next_button_1.Click
        box.a = CDbl(a_value_box.Text)
        box.b = CDbl(b_value_box.Text)
        box.c = CDbl(c_value_box.Text)

        back_button_1.Enabled = True
        next_button_1.Visible = False
        next_button_2.Visible = True
        parametries_title_label.Visible = False
        A_value_label.Visible = False
        a_value_box.Visible = False
        b_value_label.Visible = False
        b_value_box.Visible = False
        c_value_label.Visible = False
        c_value_box.Visible = False
        limit_title_label.Visible = True
        left_limit_label.Visible = True
        left_limit_box.Visible = True
        right_limit_label.Visible = True
        right_limit_box.Visible = True
    End Sub

    Private Sub back_button_1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles back_button_1.Click
        error_label.Visible = False
        back_button_1.Enabled = False
        next_button_1.Visible = True
        next_button_2.Visible = False
        parametries_title_label.Visible = True
        A_value_label.Visible = True
        a_value_box.Visible = True
        b_value_label.Visible = True
        b_value_box.Visible = True
        c_value_label.Visible = True
        c_value_box.Visible = True
        limit_title_label.Visible = False
        left_limit_label.Visible = False
        left_limit_box.Visible = False
        right_limit_label.Visible = False
        right_limit_box.Visible = False
    End Sub

    Private Sub next_button_2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles next_button_2.Click
        If (Not Cross_Points.Check_Limits(CDbl(left_limit_box.Text), CDbl(right_limit_box.Text))) Then
            error_label.Visible = True
        Else
            error_label.Visible = False
            limit_title_label.Visible = False
            left_limit_label.Visible = False
            left_limit_box.Visible = False
            right_limit_label.Visible = False
            right_limit_box.Visible = False
            back_button_2.Visible = True
            fault_label.Visible = True
            eps_label.Visible = True
            eps_box.Visible = True
            partition_label.Visible = True
            partition_box.Visible = True
            next_button_2.Visible = False
            calculate_button.Visible = True
            back_button_1.Visible = False
        End If
    End Sub

    Private Sub back_button_2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles back_button_2.Click
        error_label.Visible = False
        limit_title_label.Visible = True
        left_limit_label.Visible = True
        left_limit_box.Visible = True
        right_limit_label.Visible = True
        right_limit_box.Visible = True
        back_button_2.Visible = False
        parametries_title_label.Visible = False
        eps_label.Visible = False
        eps_box.Visible = False
        partition_label.Visible = False
        partition_box.Visible = False
        next_button_2.Visible = True
        calculate_button.Visible = False
        back_button_1.Visible = True
        fault_label.Visible = False
    End Sub

    Private Sub calculate_button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles calculate_button.Click
        If (Not (Square.Check_Partition(CInt(partition_box.Text)) And Cross_Points.Check_Epsilon(CDbl(eps_box.Text)))) Then
            error_label.Visible = True
        Else

            Cross_Points_1_2 = Cross_Points.Find_Cross_Points(CDbl(left_limit_box.Text),
                                                              CDbl(right_limit_box.Text),
                                                              CDbl(eps_box.Text), AddressOf box.F1, AddressOf box.F2)

            Cross_Points_2_3 = Cross_Points.Find_Cross_Points(CDbl(left_limit_box.Text),
                                                              CDbl(right_limit_box.Text),
                                                              CDbl(eps_box.Text), AddressOf box.F2, AddressOf box.F3)

            Cross_Points_1_3 = Cross_Points.Find_Cross_Points(CDbl(left_limit_box.Text),
                                                              CDbl(right_limit_box.Text),
                                                              CDbl(eps_box.Text), AddressOf box.F1, AddressOf box.F3)
            error_label.Visible = False
            back_button_2.Visible = False
            calculate_button.Visible = False
            print_points_button.Visible = True
            eps_label.Visible = False
            eps_box.Visible = False
            partition_label.Visible = False
            partition_box.Visible = False
            fault_label.Visible = False
        End If
    End Sub

    Private Sub print_points_button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles print_points_button.Click
        cross_point_label.Visible = True
        hint_1_2_label.Visible = True
        hint_1_3_label.Visible = True
        hint_2_3_label.Visible = True
        print_1_2_label.Text = String.Format("({0:f5},{1:f5})", Cross_Points_1_2(0).x, Cross_Points_1_2(0).y)
        print_1_2_label.Visible = True
        print_2_3_label.Text = String.Format("({0:f5},{1:f5})", Cross_Points_2_3(0).x, Cross_Points_2_3(0).y)
        print_2_3_label.Visible = True
        print_1_3_label.Text = String.Format("({0:f5},{1:f5})", Cross_Points_1_3(0).x, Cross_Points_1_3(0).y)
        print_1_3_label.Visible = True
        print_points_button.Visible = False
        area_button.Visible = True
    End Sub

    Private Sub area_button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles area_button.Click
        cross_point_label.Visible = False
        hint_1_2_label.Visible = False
        hint_1_3_label.Visible = False
        hint_2_3_label.Visible = False
        print_1_2_label.Visible = False
        print_2_3_label.Visible = False
        print_1_3_label.Visible = False
        plaza = Square.Final_Square(Cross_Points_2_3(0).x, Cross_Points_1_2(0).x, Cross_Points_1_3(0).x,
                                                      AddressOf box.F2, AddressOf box.F1, AddressOf box.F3, CDbl(eps_box.Text), CInt(partition_box.Text))
        area_button.Visible = False
        print_area_button.Visible = True
    End Sub

    Private Sub print_area_button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles print_area_button.Click
        area_label.Text = String.Format("Площадь фигуры равна {0:f8}," & vbLf & "при погрешности {1:f8}", plaza.value, plaza.error)
        area_label.Visible = True
        print_area_button.Visible = False
    End Sub
End Class
